api/src/pcapi/core/subscription/api.py

```diff
@@ -218,61 +221,200 @@ def get_profile_completion_subscription_item(
     return models.SubscriptionItem(type=models.SubscriptionStep.PROFILE_COMPLETION, status=status)
 
 
-def _get_identity_fraud_checks_for_eligibility(
-    user: users_models.User, eligibility: users_models.EligibilityType | None
-) -> list[fraud_models.BeneficiaryFraudCheck]:
-    return [
-        fraud_check
-        for fraud_check in user.beneficiaryFraudChecks
-        if fraud_check.type in fraud_models.IDENTITY_CHECK_TYPES and eligibility in fraud_check.applicable_eligibilities
-    ]
+@dataclasses.dataclass
+class UserSubscriptionState:
+    fraud_check: fraud_models.BeneficiaryFraudCheck | None = None
+    is_activable: bool = False
+    next_step: models.SubscriptionStep | None = None
+    subscription_message: models.SubscriptionMessage | None = None
+    young_status: young_status_module.YoungStatus | None = None
+
+
+@dataclasses.dataclass
+class AdminSubscriptionState(UserSubscriptionState):
+    item_status: models.SubscriptionItemStatus = models.SubscriptionItemStatus.VOID
 
 
 def get_identity_check_subscription_status(
-    user: users_models.User, eligibility: users_models.EligibilityType | None
-) -> models.SubscriptionItemStatus:
+    user: users_models.User, eligibility: users_models.EligibilityType | None = None
+) -> AdminSubscriptionState:
     """
     eligibility may be the current user.eligibility or a specific eligibility.
     The result relies on the FraudChecks made with the 3 current identity providers (DMS, Ubble and Educonnect)
     and with the legacy provider (Jouve)
     """
+    eligibility = eligibility or user.eligibility
     if eligibility is None:
-        return models.SubscriptionItemStatus.VOID
+        return AdminSubscriptionState(
+            item_status=models.SubscriptionItemStatus.VOID, young_status=young_status_module.NonEligible()
+        )
 
-    identity_fraud_checks = _get_identity_fraud_checks_for_eligibility(user, eligibility)
+    fraud_check = get_relevant_identity_fraud_check(user, eligibility)
+
+    if fraud_check:
```

Comment:
faire un "early return" si not fraud_check : ça permet de gagner un niveau d'indentation et d'améliorer la lisibilité